### ISSUE TITLE
Feature Idea: Editing comments on cards

### ISSUE THREAD

**Issue by Johennes** [mattermost/focalboard#298]
**Summary:**

It would be nice if comments were editable. Right now, once I sent a comment, I can only change it by deleting it and creating a new one.

![Screenshot 2021-04-20 at 08 38 58](https://user-images.githubusercontent.com/1137962/115349366-0c734c00-a1b4-11eb-9451-86fd6d1bf1f2.png)

**How important this is to me and why:**

Importance: Low

Low importance because deleting and recreating provides the needed functionality in most cases, only with less convenience.

Use cases:

1. I notice a typo in a comment that I just sent
2. I forgot to add something into a comment that I just sent
3. Another comment was added after the one I want to edit. Deleting and recreating/updating my comment will now change the order of comments.

**Comment by chenilim** [comment]
Yup, we want to be able to edit comments. Opening this up to any takers. Should be doable as a delete-then-insert, but might require some refactoring of the comment-list control.

**Comment by kamre** [comment]
@chenilim is this issue still actual? I would like to work on it.

In [a card to track improving comments](https://focalboard-community.octo.mattermost.com/workspace/qgsck6cts3fwpqwyjiupjm5cde/shared?id=47aa9bb4-6967-4a96-83c7-11bd6b20f1eb&v=ca1a5441-4c1c-4d0e-856e-88cc744576c8&r=762db9c6-305c-4d13-8ce6-9f11e5232c24&c=66db0d7d-11a5-4c32-83d3-ba57fc7065c6) there is a phrase: "we intend to replace the comment system with Mattermost chat". That is why I am asking about this.

**Comment by chenilim** [comment]
Hi @kamre, sorry your comment slipped through the cracks. The current plan is to keep the current comment system for v1 (i.e. not replace with Mattermost chat - I've updated [the card](https://community.mattermost.com/boards/workspace/qgsck6cts3fwpqwyjiupjm5cde/shared/47aa9bb4-6967-4a96-83c7-11bd6b20f1eb/7e38eab4-890a-4fc3-8f48-a1f9bd1ab822?r=762db9c6-305c-4d13-8ce6-9f11e5232c24&c=66db0d7d-11a5-4c32-83d3-ba57fc7065c6)). I know it's been a while, but LMK if you are still interested in taking a crack at this.

cc @wiggin77 as this might intersect with the recent `@mentions` feature, and editing might be more complicated now - i.e. we should not notify users again if editing a comment with an `@mention`, though we *should* notify newly mentioned users.

**Comment by wiggin77** [comment]
> should not notify users again if editing a comment with an `@mention`, though we _should_ notify newly mentioned users.

We already handle editing blocks, ensuring already mentioned users don't get mentioned again, but newly added ones get notified. I compare the old and new version of the block and only notify newly added mentions.
